### ISSUE TITLE
Assess / fix head marking with BinMaxHead

### ISSUE THREAD

**Issue by jbellik** [syntax-prosody-ot/main#558]
Right now binMaxHead() runs markHeads() regardless of whether heads have already been marked. This should be changed to only run if heads haven't already been marked, I think.

More importantly, the interface needs an option for selecting which side heads are aligned with for binMaxHead and binMinHead.

**Comment by jbellik** [comment]
Possibly there is a bug with not marking heads of unary things?? I am getting {(a bb ccc)} back with no heads marked as one candidate, and {(x yy zzz)} again without any heads marked with 'x yy zzz' as the input.

**Comment by jbellik** [comment]
Also {(((x*) (yy)) (zzz))} only has one head marked, when I expected all heads: {(((x*)* (yy))* (zzz*))}

**Comment by jbellik** [comment]
This may become relevant: https://github.com/syntax-prosody-ot/main/pull/562

**Comment by jbellik** [comment]
To re-present the goals of this issue, they are:

- [ ] 1. Change binMaxHead() to only run markHeads() if the tree doesn't already have heads marked. One way to implement this would be for markHeads() to add a flag to the root node: root.headsMarked = true
- [ ] 2. Check parenthesizeTree() to see if it is not translating headedness in a transparent way. A possible problem is that I've also used * to mark accentedness, I think. I suspect parenthesizeTree() is the source of the weirdness we saw on the interface, rather than markHeads() itself.
- [ ] 3. Also check markHeads() itself to see if it is buggy.
- [ ] 4. Add an option to the interface under BinMaxHead() to let the user select which side heads are marked on. Or better yet, connect the user's head side selection under GEN: Outputs to BinMaxHead. That is, if the user selects "mark heads on the left side", then pass "left" in as an option to BinMaxHead. Or if "mark heads on the right side" is selected, then pass "right" to BinMaxHead. 
- [ ] 5. Make sure the info for BinMaxHead on the interface is clear about the connection between "mark heads" and the constraint.
